flink-python/pyflink/table/table.py

```diff
@@ -0,0 +1,117 @@
+################################################################################
+#  Licensed to the Apache Software Foundation (ASF) under one
+#  or more contributor license agreements.  See the NOTICE file
+#  distributed with this work for additional information
+#  regarding copyright ownership.  The ASF licenses this file
+#  to you under the Apache License, Version 2.0 (the
+#  "License"); you may not use this file except in compliance
+#  with the License.  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+from py4j.java_gateway import get_method
+
+__all__ = ['Table']
+
+
+class Table(object):
+
+    """
+    A :class:`Table` is the core component of the Table API.
+    Similar to how the batch and streaming APIs have DataSet and DataStream,
+    the Table API is built around :class:`Table`.
+
+    Use the methods of :class:`Table` to transform data.
+
+    Example:
+    ::
+        >>> t_config = TableConfig.Builder().as_streaming_execution().set_parallelism(1).build()
+        >>> t_env = TableEnvironment.get_table_environment(t_config)
+        >>> ...
+        >>> t = t_env.scan("source")
+        >>> t.select(...)
+        ...
+        >>> t.insert_into("print")
+        >>> t_env.execute()
+
+    Operations such as :func:`~pyflink.table.Table.join`, :func:`~pyflink.table.Table.select`,
+    :func:`~pyflink.table.Table.where` and :func:`~pyflink.table.Table.group_by`
+    take arguments in an expression string. Please refer to the documentation for
+    the expression syntax.
+    """
+
+    def __init__(self, j_table):
+        self._j_table = j_table
+
+    def select(self, fields):
+        """
+        Performs a selection operation. Similar to a SQL SELECT statement. The field expressions
+        can contain complex expressions.
+
+        Example:
+        ::
+            >>> t = tab.select("key, value + 'hello'")
```

Comment:
I think using a variable to store the select result is more appropriate because select operation won't change the table itself. If we just write `tab.select("key, value + 'hello'")` and don't store its result, we will lose this select result.

Reply:
Why I suggest remove the t is because we do not use the t in the demo. See Scala version: https://github.com/apache/flink/blob/9975e0393a9e09dbde21ca61f1a5751cc5934411/flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/Table.java#L114

Reply:
For this point it makes sense. I'll adjust these document to be consistent with scala documents.